meson.build

```diff
@@ -139,6 +139,10 @@ endif
 if get_option('enable-blas')
   extra_defines += '-DUSE_BLAS=1'
 
+  if get_option('enable-neon')
+     extra_defines += '-DUSE_NEON=1'
+  endif
+
```

Comment:
You need to check the availability of neon (and others, including blas, too) at meson, before actually compiling the code.

Then, you can use "feature" instead of "boolean" for actual features (such as neon, blas, ...), which allows users to have "auto" options.

Reply:
For blas lib, we check the openblase dependency like `dependency('openblas')` and if it is not true, then return the error  before compiling. If is the openblas is false in meson_options.txt, then we just use numeric op in nntrainer itself. And for the ARM or android platform, we download the precompiled blas lib for arm from web and use it. 

However, we might need to check the target architecture and check if it supports neon (and float16 op support). 
( Moreover, openblas is necessary for the neon option case, because neon implementation only works for the division by 4 case. If the matrix is not divided by 4, then we need to run with cblas lib. This will be improved soon and then we could separate the dependency with blas )

Considering all the cases in nntrainer, I will make separate pr for the feature and auto options in meson script.